pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -446,11 +446,17 @@ func (c *MaxPDVolumeCountChecker) predicate(pod *v1.Pod, meta PredicateMetadata,
 		return false, nil, fmt.Errorf("node not found")
 	}
 
-	csiNode, err := c.csiNodeLister.Get(node.Name)
-	if err != nil {
-		// we don't fail here because the CSINode object is only necessary
-		// for determining whether the migration is enabled or not
-		klog.V(5).Infof("Could not get a CSINode object for the node: %v", err)
+	var (
+		csiNode *v1beta1storage.CSINode
+		err     error
+	)
+	if c.csiNodeLister != nil {
```

Comment:
Does the lister init also have to be feature gated? https://github.com/kubernetes/kubernetes/blob/02ee2421f54b1594aec2f0c78331e308e1b1bb8e/pkg/scheduler/framework/plugins/nodevolumelimits/csi.go#L53

Reply:
Yes, here?

https://github.com/kubernetes/kubernetes/pull/84273/files#diff-d4eb2aa05b0aaa444f6ee822fc3efed8